topostats/validation.py

```diff
@@ -0,0 +1,142 @@
+"""Validation of configuration."""
+import logging
+import os
+from pathlib import Path
+from schema import Or, Schema, SchemaError
+
+from topostats.logs.logs import LOGGER_NAME
+
+LOGGER = logging.getLogger(LOGGER_NAME)
+
+
+def validate_config(config: dict):
+    """Validate configuration.
+
+    Parameters
+    ----------
+    config: dict
+        Config dictionary imported by read_yaml() and parsed through clean_config().
+    """
+    config_schema = Schema(
```

Comment:
I suppose it could but I'm not sure how you would then define all the conditions for each field.

For example the `Or(...)` statements are not a structure I'm familiar with defining in YAML that would translate here on reading.

To an extent though, its a bit chicken and egg, there is `topostats/default_config.yaml` that we could import and test and in a pending PR (#307) I've documented copying that and editing it, but if someone edits it locally and its read in and used as the Schema then their modification will always pass.